diff -up gromox-2.30/Makefile.am.omv~ gromox-2.30/Makefile.am
--- gromox-2.30/Makefile.am.omv~	2024-08-08 19:01:55.095428981 +0200
+++ gromox-2.30/Makefile.am	2024-08-08 19:03:05.548233915 +0200
@@ -251,6 +251,7 @@ timer_LDADD = -lpthread ${libHX_LIBS} li
 
 libphp_mapi_la_CPPFLAGS = ${AM_CPPFLAGS} ${PHP_INCLUDES}
 libphp_mapi_la_SOURCES = php_mapi/ext.hpp php_mapi/ext_pack.cpp php_mapi/mapi.cpp php_mapi/mapi_arginfo.hpp php_mapi/rpc_ext.cpp php_mapi/type_conversion.cpp php_mapi/type_conversion.hpp php_mapi/zclient.cpp php_mapi/zrpc.cpp
+libphp_mapi_la_LDFLAGS = -Wl,--allow-shlib-undefined
 libphp_mapi_la_LIBADD = ${fmt_LIBS} ${libHX_LIBS} libgromox_common.la libgromox_mapi.la
 libmapi4zf_la_SOURCES =
 libmapi4zf_la_LDFLAGS = ${PHP_LDFLAGS}
